### PR TITLE
bupstash 0.10.2 (new formula)

### PR DESCRIPTION
**Pull request by bket**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/bupstash.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  test do
+    (testpath/"testfile").write("This is a test")
+
+    system "#{bin}/bupstash", "init", "-r", testpath/"foo"
+    system "#{bin}/bupstash", "new-key", "-o", testpath/"key"
+    system "#{bin}/bupstash", "put", "-k", testpath/"key", "-r", testpath/"foo", testpath/"testfile"
+
+    assert_equal shell_output("cat #{testpath}/testfile"),
```
Comment (dawidd6) [inline Formula/bupstash.rb:33]:
```suggestion
    assert_equal (testpath/"testfile").read
```

Comment (bket) [reply Formula/bupstash.rb]: @dawidd6, branch has been force pushed with you suggestion. Thank you!

File: Formula/bupstash.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+class Bupstash < Formula
+  desc "Easy and efficient encrypted backups"
+  homepage "https://bupstash.io"
+  url "https://github.com/andrewchambers/bupstash/releases/download/v0.10.2/bupstash-v0.10.2-src+deps.tar.gz"
```
Comment (carlocab) [inline Formula/bupstash.rb:4]: What deps does this bundle?

File: Formula/bupstash.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      man7.install Dir["*.7"]
+    end
+  end
+
+  test do
+    (testpath/"testfile").write("This is a test")
+
+    system "#{bin}/bupstash", "init", "-r", testpath/"foo"
+    system "#{bin}/bupstash", "new-key", "-o", testpath/"key"
+    system "#{bin}/bupstash", "put", "-k", testpath/"key", "-r", testpath/"foo", testpath/"testfile"
```
Comment (carlocab) [inline Formula/bupstash.rb:31]:
```suggestion
    system bin/"bupstash", "init", "-r", testpath/"foo"
    system bin/"bupstash", "new-key", "-o", testpath/"key"
    system bin/"bupstash", "put", "-k", testpath/"key", "-r", testpath/"foo", testpath/"testfile"
```
This is a little cleaner, I think.

**Review by carlocab — COMMENTED** [review]
Thanks, @bket. Just one question and a minor suggestion.

File: Formula/bupstash.rb (hunk 2 above)
Comment (bket) [reply Formula/bupstash.rb:4]: @carlocab, I just force pushed with your suggestion. Thank you!

> What deps does this tarball bundle?

The tarball bundles 96 cargo crates, most of which have not been packaged by homebrew:

```
ahash				indicatif			rusqlite
aho-corasick			instant				ryu
anyhow				itoa				same-file
arrayref			lazy_static			scopeguard
arrayvec			libc				serde
atty				libsqlite3-sys			serde_bare
autocfg				lock_api			serde_derive
bitflags			lz4				serde_json
blake3				lz4-sys				serial_test
cc				memchr				serial_test_derive
cfg-if				nix				shlex
cfg-if-0.1.10			num-integer			smallvec
chrono				num-traits			syn
codemap				number_prefix			tar
codemap-diagnostic		once_cell			tempfile
console				openat				termcolor
constant_time_eq		parking_lot			terminal_size
crossbeam-channel		parking_lot_core		thiserror
crossbeam-utils			path-clean			thiserror-impl
digest				pkg-config			time
encode_unicode			ppv-lite86			typenum
fallible-iterator		proc-macro2			unicode-width
fallible-streaming-iterator	quote				unicode-xid
filetime			rand				vcpkg
generic-array			rand_chacha			version_check
getopts				rand_core			walkdir
getrandom			rand_hc				wasi
glob				rangemap			winapi
hashbrown			redox_syscall			winapi-i686-pc-windows-gnu
hashlink			regex				winapi-util
hermit-abi			regex-syntax			winapi-x86_64-pc-windows-gnu
humantime			remove_dir_all			xattr
```

File: Formula/bupstash.rb (hunk 3 above)
Comment (bket) [reply Formula/bupstash.rb]: @carlocab, I just force pushed with your suggestion. Thank you!

**Review by carlocab — APPROVED** [review]
Thanks, @bket.

**Pull request closed without merging**